### PR TITLE
Installer

### PR DESCRIPTION
**Pull request by harikt**
I think I have covered the stuffs. 

Let me know if I missed something.

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,10 +10,34 @@ It's totally not the first approach of this kind. But: this one should be easily
...
-    $ wireshell new
+$ wireshell new /path/where/to/install --dbUser=db-user --dbName=db-name --dbPass=db-password --httpHosts=pw.dev --adminUrl=processwire --username=admin --userpass=abcd1234 --useremail=someone@example.com
+```
+
+If you don't pass the values, it will ask interactively.
+
+#### Profile installation
+
+```
+$ wireshell one-click-install /path/where/to/install --dbUser=db-user --dbName=db-name --dbPass=db-password --httpHosts=pw.dev --adminUrl=processwire --username=admin --userpass=abcd1234 --useremail=someone@example.com --profile=/path/to.zip
```
Comment (marcus-herrmann) [inline README.md]: The install command is still called "one-click-install" here.

File: README.md [hunk 2]
```diff
@@ -10,10 +10,34 @@ It's totally not the first approach of this kind. But: this one should be easily
...
+    site-default/
+        modules
+        templates
+    composer.json
+```
+
+#### Download only
+
+```
+    $ wireshell new path --no-install
```
Comment (marcus-herrmann) [inline README.md]: Please use `/path/where/to/install` in this example like above.

File: src/Commands/NewCommand.php [hunk 3]
```diff
@@ -18,187 +42,594 @@
...
+            ->addOption('timezone', null, InputOption::VALUE_REQUIRED, 'Timezone.')
+            ->addOption('chmodDir', null, InputOption::VALUE_REQUIRED, 'Directory mode. Defaults 755.')
+            ->addOption('chmodFile', null, InputOption::VALUE_REQUIRED, 'File mode. Defaults 644')
+            ->addOption('httpHosts', null, InputOption::VALUE_REQUIRED, 'Hostname without www part.')
+            ->addOption('adminUrl', null, InputOption::VALUE_REQUIRED, 'Admin url.')
+            ->addOption('username', null, InputOption::VALUE_REQUIRED, 'Admin username.')
+            ->addOption('userpass', null, InputOption::VALUE_REQUIRED, 'Admin password.')
+            ->addOption('useremail', null, InputOption::VALUE_REQUIRED, 'Admin email address.')
+            ->addOption('profile', null, InputOption::VALUE_REQUIRED, 'Default site profile.')
+            // ->addArgument('color', null, InputOption::VALUE_NONE, 'Default ProcessWire color.')
```
Comment (marcus-herrmann) [inline src/Commands/NewCommand.php]: Please remove not used ->addArgument like.

File: src/Commands/NewCommand.php [hunk 4]
```diff
@@ -18,187 +42,594 @@
...
+        $this->fs = new Filesystem();
+        $directory = rtrim(trim($input->getArgument('directory')), DIRECTORY_SEPARATOR);
+        $this->projectDir = $this->fs->isAbsolutePath($directory) ? $directory : getcwd().DIRECTORY_SEPARATOR.$directory;
+        $this->projectName = basename($directory);
+
+        $logger = new Logger('name');
+        $logger->pushHandler(new StreamHandler("php://output"));
+        $this->installer = new Installer($logger, $this->projectDir);
+
+        // $this->version = trim($input->getArgument('version'));
```
Comment (marcus-herrmann) [inline src/Commands/NewCommand.php]: Please remove unused line

File: src/Commands/NewCommand.php [hunk 5]
```diff
@@ -18,187 +42,594 @@
...
+                }
+                $accountInfo['useremail'] = $useremail;
+
+    /*
+            $colors = array(
+                'classic',
+                'warm',
+                'modern',
+                'futura'
+            );
```
Comment (marcus-herrmann) [inline src/Commands/NewCommand.php]: Please remove comment / unused array.

File: src/Commands/NewCommand.php [hunk 6]
```diff
@@ -18,187 +42,594 @@
...
+        $this->output->writeln("\n Downloading ProcessWire...");
+
+        // decide which is the best compressed version to download
+        $distill = new Distill();
+        // $branch = 'https://github.com/ryancramerdesign/ProcessWire/archive/master.zip';
+        $pwArchiveFile = $distill
+            ->getChooser()
+            ->setStrategy(new MinimumSize())
+            ->addFile($branch)
+            // ->addFilesWithDifferentExtensions($branch, ['zip', 'tgz'])
```
Comment (marcus-herrmann) [inline src/Commands/NewCommand.php]: Please remove comment / unused lines.

File: composer.json [hunk 7]
```diff
@@ -17,6 +17,10 @@
   "require": {
     "php": ">=5.4.0",
     "symfony/console": "~2.0",
-    "guzzlehttp/guzzle": "~4.0"
+        "guzzlehttp/guzzle": "~4.0",
```
Comment (marcus-herrmann) [inline composer.json:20]: You removed GuzzleHttp in the wireshell executable, it also can be removed from composer.json.

File: README.md [hunk 8]
```diff
@@ -10,10 +10,34 @@ It's totally not the first approach of this kind. But: this one should be easily
...
+
+You can also install profiles. Current structure of zip is as
+
+```
+myprofile/
+    site-default/
+        modules
+        templates
+    composer.json
+```
```
Comment (marcus-herrmann) [inline README.md:35]: Could you explain why a composer.json is needed here? Thanks!

File: README.md [hunk 9]
```diff
@@ -10,10 +10,34 @@ It's totally not the first approach of this kind. But: this one should be easily
 
 Currently, Wireshell  consists of the following basic commands:
 
-#### New
+#### Fresh installation
```
Comment (marcus-herrmann) [inline README.md:13]: Could you please structure this section as follows:

```
### New
...

##### Fresh installation
...

##### Profile installation
...

##### Download only
...

#### Create-user
...
```

Cheers!

File: src/Commands/NewCommand.php (hunk 4 above)
Comment (harikt) [inline src/Commands/NewCommand.php]: Hi @marcus-herrmann , the idea is at some point of time when PW add tags to all release, we can pass and get the version. I like it to stay there, if you are very strong on it, will remove the same when I am back.

File: composer.json (hunk 7 above)
Comment (harikt) [inline composer.json:20]: nope. It is still used inside the command.

File: README.md (hunk 8 above)
Comment (harikt) [inline README.md:35]: That is if there is a composer.json install the dependencies from it. You can see in the Installer command it tries to execute the `composer install` via symfony/process.

**Comment by harikt** [comment]
Will try to address your comments when I am back.

Moving out for a while.

File: README.md (hunk 8 above)
Comment (marcus-herrmann) [inline README.md:35]: I see!

File: composer.json (hunk 7 above)
Comment (marcus-herrmann) [inline composer.json:20]: But it isn' present anymore in here: https://raw.githubusercontent.com/harikt/wireshell/installer/wireshell

**Comment by marcus-herrmann** [comment]
Thanks again! :+1:

File: composer.json (hunk 7 above)
Comment (harikt) [inline composer.json:20]: See here https://github.com/harikt/wireshell/blob/installer/src/Commands/NewCommand.php#L357

Comment (marcus-herrmann) [inline composer.json:20]: Right - nevermind my comment.

File: README.md (hunk 9 above)
Comment (harikt) [inline README.md:13]: I have changed, may be you make sure the thing I did is what you asked.

Thank you

Comment (marcus-herrmann) [inline README.md:13]: That was not exactly what I meant, but anyway, it's just a detail and I'll merge (and maybe modify later)

Comment (harikt) [inline README.md:13]: Thank you

_Hari K T_

You can ring me : +91 9388 75 8821

http://harikt.com , https://github.com/harikt ,
http://www.linkedin.com/in/harikt , http://www.xing.com/profile/Hari_KT

Skype  : kthari85
Twitter : harikt

Comment (marcus-herrmann) [inline README.md:13]: :+1:

**Pull request merged**